File: Swift/DemoDispatchOC/DemoDispatchOC/ViewController.m
//
//  ViewController.m
//  DemoDispatchOC
//
//  Created by hrt03 on 2019/11/15.
//  Copyright © 2019 dengchaojie. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
@property (nonatomic, strong) UIImage *img;

@end

@implementation ViewController
//https://avatars2.githubusercontent.com/u/13744851?s=400&u=acd2cf709c9b572d6c5a53b1a728a7cba276355c&v=4
- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    UIImageView *imgV = UIImageView.new;
    [self.view addSubview:imgV];
    self.img = [UIImage imageWithData:[NSData dataWithContentsOfURL:[NSURL URLWithString:@"https://b-gold-cdn.xitu.io/v3/static/img/45.b99ea03.svg"]]];
    imgV.frame = CGRectMake(100, 100, 100, 100);
    
    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(2 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
        imgV.image = self.img;
    });
    
}



@end

File: Swift/DemoDispatchStoryboard/DemoDispatchStoryboard/ViewController.swift
//
//  ViewController.swift
//  DemoDispatchStoryboard
//
//  Created by chao jie deng on 2019/11/15.
//  Copyright © 2019 chao jie deng. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
        let imgV = UIImageView.init(frame: CGRect.init(x: 100, y: 100, width: 100, height: 100))
        self.view.addSubview(imgV)
        
        do {
             imgV.image = UIImage.init(data: try Data.init(contentsOf: URL.init(string: "https://b-gold-cdn.xitu.io/v3/static/img/45.b99ea03.svg")!))
            print("main111")
        } catch  {
            print("error")
        }
//        didTapOnGenerateImageButton()

        
    }


}

func didTapOnGenerateImageButton() {
    
    DispatchQueue.global(qos: .background).async {
        do {
            let data = try Data.init(contentsOf: URL.init(string: "https://b-gold-cdn.xitu.io/v3/static/img/45.b99ea03.svg")!)
            
            
            DispatchQueue.main.async {
                let image = UIImage.init(data: data)
                print(image as Any)
                print("main")

            }
        } catch  {
            print("error")
        }
        
    }
}
